src/border.jl

```diff
@@ -902,7 +902,12 @@ function padarray(::Type{T}, img::AbstractArray{S,N}, f::Fill{_,N}) where {T,S,_
     catch
         error("Unable to fill! an array of element type $(eltype(A)) with the value $(f.value). Supply an appropriate value to `Fill`, such as `zero(eltype(A))`.")
     end
-    A[indices(img)...] = img
+
+    # We would have liked to do A[indices(img)...] = img, but this calls size(img)
+    # which is not defined if img is of type OffsetArrays. 
+    for i in collect(Iterators.product(OffsetArrays.axes(img)...))
+             setindex!(A,getindex(img,i...),i...)
+    end
```

Comment:
How about 

```julia
for I in CartesianRange(indices(img))
    A[I] = img[I]
end
```

Reply:
I used `@which Compat.axes(img)` for both a traditional array and Offset array and it seems to be calling the same functions as `OffsetArrays.axes(img)`. Why do we use `Compat.axes(img)` in this instance and not `OffsetArrays.axes(img)`? I thought the `Compat` package has something to do with deprecations?